core/src/main/java/com/pholser/junit/quickcheck/runner/Shrinker.java

```diff
@@ -56,7 +56,7 @@ a copy of this software and associated documentation files (the
         this.testClass = testClass;
         this.failure = failure;
         this.maxShrinks = shrinkControl.maxShrinks();
-        this.maxShrinkDepth = shrinkControl.maxShrinkDepth();
+        this.maxShrinkDepth = shrinkControl.maxShrinkDepth() * method.getMethod().getParameterCount();
```

Comment:
(I could easily have missed this in the code)
is max shrink depth handled the same for multiple params? That is, if max depth = 50 for a property with 2 params, and both params will generate shrinks with depth 150, do param 1 and param 2 each get a depth 50 search?

Reply:
`maxDepth` is a per-property configuration, rather than a per-parameter configuration. In cases of more than one property parameter, we have `maxDepth` multiplied by number of parameters to allow for greater tree depth.